### DISCUSSION TITLE
Issues getting set up - (compiling?)

### DISCUSSION THREAD

**Discussion by davet1234** [awawa-dev/HyperSerialEsp8266#10] [Q&A]
Hi,

I'm having a little trouble getting HyperSerial set up.
I have a Raspberry Pi 4b running HyperHDR 17, an ESP826 (CH340G) flashed with HyperSerialEsp8266, and a strip of WS2812b LEDs (206 LEDs). I have followed the instructions up to the "Compiling" section, but I'm not sure what I'm supposed to be doing there. Are there any more detailed instructions anywhere that I can follow? Or a tutorial that you're aware of?

Thanks!

**Comment by awawa-dev** [comment]
Hi
Provided bin images should be sufficient (flash them using for example esphome-flasher) unless you want to make some changes in the project. Compiling from sources is rather for advanced users: in fact it's typical path for generic nodeMCU project including installing one external library so you can follow any Arduino Studio tutorial for ESp8266 projects. I guess the most tricky part for most  users is to install correct CH340G drivers under Windows and to add Esp8266 board to Arduino Studio, but again: you can find tone of such tutorials on the net...you need to be more specific.

**Reply by davet1234** [reply]
Thanks for getting back to me. I was hoping that was the case, but the issue I'm having is that the LEDs aren't coming on - just the first two light up. The strip definitely works as I can get it all working through WLED, but HyperSerial seems like a much better option if I can get it to work!

**Reply by awawa-dev** [reply]
And that two lights react correctly to color that you set in HyperHDR. Or they are just some random lights? Frankly there are not much options to change in HyperSerialEsp8266 sketch for WS2812b except speed...but that would mean that you have counterfeited CH340G. You can try to test HyperSerialWLED: first 2Mb speed version to make sure the serial chip is capable of that speed. If won't work there is also special version of HyperSerialWLED that works at 921600 (you must set the same speed in HyperHDR, AWA protocol must be enabled for all versions of HyperSerial).

**Reply by davet1234** [reply]
The first two lights don't react to the picture on the tv.
I've tried HyperSerialWLED and it doesn't work (the 921600 version works great, but I'd rather avoid WLED if possible) - does that mean I have a counterfeit ch340g? Is there any other way to tell if it's counterfeit? I'll happily buy another one if it's going to make this work, but I want to avoid buying another fake if possible!

**Reply by awawa-dev** [reply]
For 99% ESP doesn't receive any data because, it can't work at such speed (or wrong configuration in HyperHDR or sketch was not for ws2812b) and you've got just some ghost lights. You should rather do testing using solid colors from the HyperHDR remote/effects tab than grabber. One important thing: did you try HyperSerialWLED@921600 version? For most users having counterfeit 340G that version works fine. We need to exclude other possible causes like wrong HyperHDR configuration or bad USB cable. Did LED react for HyperSerialWLED@921600 commands properly when you set the color using WLED WWW page, but HyperSerialWLED@921600 was not receiving any data from HyperHDR when you tried it?

**Reply by davet1234** [reply]
The 921600 version works absolutely perfectly with HyperHDR and through the WLED interface. Does that mean it's most likely to be a counterfeit CH340g?

**Reply by awawa-dev** [reply]
Yes, unfortunately that's the case. Genuine CH340g has no problems with 2000000 speed using Adalight USB mode.

**Reply by davet1234** [reply]
Well that's solved that - thanks so much for your help; I really appreciate it! And thanks for making HyperHDR - it's a big improvement over Hyperion! 👍
